dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheHttpClientBlockingChannel.java

```diff
@@ -70,6 +70,8 @@ public Response execute(Endpoint endpoint, Request request) throws IOException {
         if (request.body().isPresent()) {
             Preconditions.checkArgument(
                     endpoint.httpMethod() != HttpMethod.GET, "GET endpoints must not have a request body");
+            Preconditions.checkArgument(
+                    endpoint.httpMethod() != HttpMethod.HEAD, "HEAD endpoints must not have a request body");
```

Comment:
should we have similar validation to check people don't start relying on response bodies from HEAD requests?

Reply:
Updated with tests.